src/stdlib_linalg.fypp

```diff
@@ -108,4 +201,197 @@ contains
+      end function is_diagonal_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      pure function is_symmetric_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        integer :: A_shape(2), n, i, j
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be symmetric
+        end if
+        A_shape = shape(A)
+        n = A_shape(1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j-1 !loop over all rows above diagonal
+                if (.not. (A(i,j) .eq. A(j,i))) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is symmetric
+      end function is_symmetric_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      pure function is_skew_symmetric_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        integer :: A_shape(2), n, i, j
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be skew-symmetric
+        end if
+        A_shape = shape(A)
+        n = A_shape(1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j !loop over all rows above diagonal (and diagonal)
+                if (.not. (A(i,j) .eq. -A(j,i))) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is skew-symmetric
+      end function is_skew_symmetric_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      pure function is_hermitian_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        integer :: A_shape(2), n, i, j
+        if (.not. is_square(A)) then
+           res = .false.
+           return !nonsquare matrices cannot be Hermitian
+        end if
+        A_shape = shape(A)
+        n = A_shape(1) !symmetric dimension of A
+        do j = 1, n !loop over all columns
+            do i = 1, j !loop over all rows above diagonal (and diagonal)
+                if (.not. (A(i,j) .eq. conjg(A(j,i)))) then
+                  res = .false.
+                  return
+                end if
+            end do
+        end do
+        res = .true. !otherwise A is Hermitian
+      end function is_hermitian_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      function is_triangular_${t1[0]}$${k1}$(A,uplo) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        character, intent(in) :: uplo
+        logical :: res
+        ${t1}$ :: zero
+        integer :: A_shape(2), m, n, o, i, j
+        zero = 0 !zero of relevant type
+        A_shape = shape(A)
+        m = A_shape(1)
+        n = A_shape(2)
```

Comment:
```suggestion
        integer :: m, n, o, i, j
        zero = 0 !zero of relevant type
        m = size(A,1)
        n = size(A,2)
```

I think it's fine to get rid of one variable here.

Reply:
I agree, thanks for pointing that out. I'm not going to commit your suggestion just because I have to fix it in multiple places/functions (which I'll do).